### PR TITLE
71 create login logout pages

### PR DESCRIPTION
**Pull request by fadeomar**
#### Issues this PR relates to
#71
=======

_NOTE: REMEMBER NOT TO REVIEW UNLESS ALL ACCEPTANCE CRITERIA HAS BEEN TICKED OFF_

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist

Please make sure you ensure the following has been done before marking your PR as awaiting-review. 

- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/components/Inputs/BasicInput/index.js [hunk 1]
```diff
@@ -79,6 +81,17 @@ const BasicInput = ({
...
+        <T.Link
+          to={GENERAL.FORGET_PASSWORD}
+          color="neutralDark"
+          style={{ display: 'block' }}
+          mt="3"
+          ml="2"
+        >
+          Forget password?
+        </T.Link>
+      )}
```
Comment (RamyAlshurafa) [inline client/src/components/Inputs/BasicInput/index.js:94]: keep this out of the common component please, we agreed to make common components small and less logic inside

File: client/src/pages/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      }
+    } else {
+      // after that the user should be directed to its dashboard
+      history.push(R.ADMIN.HOME);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setState({ submitAttempt: true });
```
Comment (RamyAlshurafa) [inline client/src/pages/Login/index.js:77]: use `useRef` instead of set state, as this doesn't require to dom re-render

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]

File: client/src/pages/Login/index.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            <T.P mb="2" color="error">
+              {httpError}
+            </T.P>
+          </Col>
+        )}
+        <Col w={[4, 11, 6]} style={{ alignItems: 'flex-end' }}>
+          <Button
+            variant="primary"
+            disabled={false}
+            loading={false}
```
Comment (RamyAlshurafa) [inline client/src/pages/Login/index.js:128]: handle this please

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
image

also when got http error, the style went like this

File: client/src/components/Inputs/BasicInput/index.js (hunk 1 above)
Comment (fadeomar) [reply client/src/components/Inputs/BasicInput/index.js:94]: done

File: client/src/pages/Login/index.js (hunk 2 above)
Comment (fadeomar) [reply client/src/pages/Login/index.js:77]: done

File: client/src/pages/Login/index.js (hunk 3 above)
Comment (fadeomar) [reply client/src/pages/Login/index.js:128]: done

Comment (RamyAlshurafa) [reply client/src/pages/Login/index.js:146]: i can't see the fix

File: client/src/pages/Login/index.js [hunk 4]
```diff
@@ -106,22 +114,32 @@ const Login = () => {
...
-      <Row mt="7" mtT="6" style={{ flex: 1, alignItems: 'flex-end' }}>
-        {httpError && (
-          <Col w={[4, 12, 12]}>
+      <Row
+        mt="7"
+        mtT="6"
+        style={{ flex: Number(isMobile), alignItems: 'flex-end' }}
+      >
+        <Col w={[4, 11, 6]} style={{ alignItems: 'flex-end' }}>
+          {!httpError && (
```
Comment (RamyAlshurafa) [inline client/src/pages/Login/index.js:138]: i think the check is reveresed

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]
the forget password link causes a hard reload
https://www.loom.com/share/7165da0c840943ba996984c28aaf64d0

**Review by RamyAlshurafa — APPROVED** [review]

**Pull request merged**